### ISSUE TITLE
Allow a list to be scrolled with the window's scroll event

### ISSUE THREAD

**Issue by helfi92** [bvaughn/react-window#135]
It would be great to allow a list be scrolled with the window's scroll event. It doesn't have to be the default behavior. We could expose a window scroll prop which would make it an opt-in feature. Thoughts?

**Comment by bvaughn** [comment]
This sounds like a duplicate of #30 so I'm going to just copy-paste my answer from that issue since it still applies 😄 

---

`WindowScroller` was one of those painful to maintain components. I see the usefulness of it, but I'm not sure I'm interested in committing to maintain another one. I would prefer instead, if possible, for an external component to be built (and published separately) that implements this functionality on top of react-window.

I might be willing to make a couple of changes to react-window to support such a component, if it existed, depending on how complex they would be.

---

If you're willing to put together a proposal (demo code, proposed API changes) I'm willing to talk it over and consider potential, minor changes to this library– but I'm confident that I'm not interested in the following at this time:
* Adding a `WindowScroller` (or similar) component to the react-window package. I don't have the time or desire to take on the maintenance burden of such a component.
* Adding "controlled" props for `scrollLeft` and `scrollTop`. That sort of an API (mix of controlled an uncontrolled props for scroll offset) [was a big regret of mine with react-virtualized](https://twitter.com/brian_d_vaughn/status/959600888242307072?lang=en. It's best to either go fully controlled (no state, only props) or fully uncontrolled (only state, with maybe a `props.defaultValues` that's read only once before mounting). react-window already has an uncontrolled API for this (with imperative methods for setting an overridden value).

I'm going to close this issue now because I don't plan on taking any direct action, but we can continue the discussion here 😄
